### ISSUE TITLE
SubstituteParameters rule should detect unescaped & in string

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#1185]
```
DEFINE VARIABLE lcFoo AS CHARACTER NO-UNDO.
lcFoo = SUBSTITUTE("test &1 & test":U,"foo":U)
```

This creates a runtime error because the & is not escaped. The rule (or a new one?) should check this.

```
Invalid character '<character>' following '&' in 
SUBSTITUTE. (2920)
The control string in the SUBSTITUTE function 
contains an invalid character following an '&' 
reference. The only valid characters are the values 
'1' through '9', indicating which following 
parameter value should be substituted at that 
point. To include a plain '&' in the control string, 
use a double ampersand ("&&"). "
```

**Comment by gquerret** [comment]
Confirmed, will be implemented.
